DOCS_GUIDELINE.md

```diff
@@ -0,0 +1,72 @@
+# Documentation guideline
+
+## General guidelines
+
+* Always use `hexadecimal` when defining offsets. They can be defined as follows. 
+  * With `0x` before the value: `0x1234`
+  * With `h` after the value: `1234h`
+
+* Although `hexadecimal` is preferred, `decimal` can be used for the rest of values.
+
+---
+
+## Representing single variables
+
+Here's a list of the recommended basic variable types. Size is in *bits*.
+
+| Size | Signed | Unsigned
+|------|--------|----------
+| 8  |  int8  |  uint8  
+| 16 |  int16 |  uint16
+| 32 |  int32 |  uint32
+| 32 |  float |  -
+| 64 |  int64 |  uint64
+| 64 |  double |  -
+
+If the variable contains actual text, `char` is preferred to `uint8` or `int8`.
+
+Arrays must be represented with the **[]** symbol. For example `int8[4]` means 4 int8 variables.
+The number of items in an array must be defined in `decimal` unless it is much more convenient to read in `hexadecimal`.
+
+Here are a few other special types you should take in account:
+
+| Size | Name | 
+|------|--------|
+| 64 |  Vector2f  |   
+| 96 |  Vector3f |  
+| 128 |  Vector4f |  
+| 512 |  Matrix4x4 |  
+
+---
+
+## Representing bitfields
+
+Bitfields always have to have their relative position and field size expressed in **decimal**.
```

Comment:
A reason for this to be decimal? Positions are better presented by hexadecimal, wouldn't you think?

Reply:
Bitfields will rarely go over 32 bits in total and 99% of programming definitions of bitfields use decimal to specify the length.
For example:

```
struct bField
{
   int a:1;
   int b:8;
   int c:7;
}
```

So, I think using decimal makes more sense.

Reply:
Oh I see. Very well then.